include/nbl/builtin/hlsl/cpp_compat/intrinsics.h

```diff
@@ -10,6 +10,7 @@
 #include <algorithm>
 #include <cmath>
 #include "nbl/core/util/bitflag.h"
+#include <nbl/builtin/hlsl/dot_product.hlsl>
```

Comment:
if you don't want it in the same header, then move into folder `intrinsics/dot_product` otherwise we'll get clutter

Reply:
its not that i don't want it in the same header, `intrinsics.h` is C++ only header so i need another header to use `nbl::hlsl::dot` in HLSL